### ISSUE TITLE
Something's adding spaces to MIME parameters

### ISSUE THREAD

**Issue by pshaughn** [servo/servo#24918]
This could have the same eventual resolution as #24874. WPT is expecting `"yo/yo;charset=UTF-8/yo=YO;x=y"` but sees `"yo/yo; charset=utf-8; yo=YO; x=y`  with the charset (and ONLY the charset) forced to lowercase and spaces added between the parameters.

**Comment by pshaughn** [comment]
xhr/send-content-type-charset has lots of failing subtests. In some a string is getting changed when it should stay verbatim, and in others like `text/plain;charset= utf-8` (whitespace _sic_) the test expects a change that doesn't happen.

**Comment by pshaughn** [comment]
This may be a point of browser disagreement, looking at https://wpt.fyi/results/xhr/send-content-type-charset.htm?label=experimental&label=master&aligned .

**Comment by jdm** [comment]
This may be related to #20436.

**Comment by pshaughn** [comment]
Yes, both are about entrusting a string to a Hyper data structure and incorrectly expecting to see the same string again later when we ask for it. We need to be much stringier about MIME types if we want to pass the WPT tests that other browsers pass.
